Userland/Libraries/LibAudio/WavLoader.cpp

```diff
@@ -185,16 +187,44 @@ bool WavLoaderPlugin::parse_header()
     CHECK_OK("Sample rate");
 
     read_u32();
-    CHECK_OK("Byte rate");
+    CHECK_OK("Data rate");
 
     read_u16();
-    CHECK_OK("Block align");
+    CHECK_OK("Block size");
 
-    m_bits_per_sample = read_u16();
+    u16 bits_per_sample = read_u16();
     CHECK_OK("Bits per sample"); // incomplete read check
-    ok = ok && (m_bits_per_sample == 8 || m_bits_per_sample == 16 || m_bits_per_sample == 24);
-    CHECK_OK("Bits per sample"); // value check
-    VERIFY(m_bits_per_sample == 8 || m_bits_per_sample == 16 || m_bits_per_sample == 24);
+    // PCM can only have 16 or 24-bit samples
+    if (audio_format == WAVE_FORMAT_PCM) {
+        ok = ok && (bits_per_sample == 8 || bits_per_sample == 16 || bits_per_sample == 24);
+        CHECK_OK("Bits per sample (PCM)"); // value check
+        VERIFY(bits_per_sample == 8 || bits_per_sample == 16 || bits_per_sample == 24);
+    }
+    // Float can only have 32 or 64-bit samples
+    else if (audio_format == WAVE_FORMAT_IEEE_FLOAT) {
+        ok = ok && (bits_per_sample == 32 || bits_per_sample == 64);
+        CHECK_OK("Bits per sample (Float)"); // value check
+        VERIFY(bits_per_sample == 32 || bits_per_sample == 64);
+    }
+
+    // Depending on sample size and audio format, determine sample format
+    if (audio_format == WAVE_FORMAT_PCM && bits_per_sample == 8) {
+        m_sample_format = PcmSampleFormat::Uint8;
+    } else if (audio_format == WAVE_FORMAT_PCM && bits_per_sample == 16) {
+        m_sample_format = PcmSampleFormat::Int16;
+    } else if (audio_format == WAVE_FORMAT_PCM && bits_per_sample == 24) {
+        m_sample_format = PcmSampleFormat::Int24;
+    } else if (audio_format == WAVE_FORMAT_IEEE_FLOAT && bits_per_sample == 32) {
+        m_sample_format = PcmSampleFormat::Float32;
+    } else if (audio_format == WAVE_FORMAT_IEEE_FLOAT && bits_per_sample == 64) {
+        m_sample_format = PcmSampleFormat::Float64;
+    } else {
+        VERIFY_NOT_REACHED();
+    }
```

Comment:
You can make this in less comparissons, you are checking the same thing multiple times

Reply:
Still verifying twice, that the bits_per_sample is valid

Reply:
This follows the style of the other code in this function, and I think it's outside of the scope of this PR.